content/issues/1/weaving-interface/annotated-zoom.js

```diff
@@ -0,0 +1,72 @@
+/// todos
+// - disable scroll when positioning contextual note
+// - position contextual note better vertically
+// - disable back link in notes? (doesn't work now / not sure how to make work)
+// - fix or remove broken images
+// - link to books on derrida's margins site?
+
+
+window.addEventListener('DOMContentLoaded', (event) => {
+
+    // load annotation overlays and then use to initialize viewer
+    fetch('annotations.json')
+      .then(response => response.json())
+      .then(data => initAnnotatedZoom(data));
+
+    function initAnnotatedZoom(data) {
+
+        // create elements with page numbers to be positioned as overlays
+        var pageDiv = document.getElementById('page_numbers');
+        data.overlays.filter(el => el.className == 'page_n')
+            .forEach(el => {
+                const pageSpan = document.createElement('span');
+                pageSpan.innerText = el.content;
+                pageSpan.setAttribute('id', el.id);
+                pageDiv.appendChild(pageSpan);
+            });
+
+        var viewer = OpenSeadragon({
+            id:"dataweaving-zoom",
+            prefixUrl:"https://cdn.jsdelivr.net/npm/openseadragon@2.4/build/openseadragon/images/",
+            preserveViewport: true,
+            visibilityRatio: 1,
+            // disallow zoom in/out
+            minZoomLevel: 4,
+            maxZoomLevel: 4,
+            defaultZoomLevel: 4,
+            // disable vertical panning
+            panVertical: false,
```

Comment:
this combination of settings for me resulted in not seeing the numbers above (or footnotes below) the weaving on page load - I only discovered them because I used the red box in the preview overlay to pan around.

Reply:
Yes, I know it doesn't look right at some screen sizes  — my plan was to work on that with @gissoo and add some media queries to set the appropriate zoom level for the screen size.

Gissoo had proposed having a slider to control zooming in and out, which I was able to prototype, but there were some complications with getting it to work with the other controls and interactions. My inclination is to keep the interaction simpler if we can make it work.